### PR TITLE
Fixing error after playing music several times with AudioCache

### LINKED ISSUE #843: Error after playing music several times with AudioCache
Error after playing music several times with AudioCache.

**Full Description**

By executing the following code, an error occurs when press the button several times.

I found that there are already related issues:

#773 #682 #766 

I think the reason of the issue is at `play` method of `AudioCache`.

```
Future<AudioPlayer> play(
    String fileName, {
    double volume = 1.0,
    bool? isNotification,
    PlayerMode mode = PlayerMode.MEDIA_PLAYER,
    bool stayAwake = false,
    bool recordingActive = false,
    bool? duckAudio,
  }) async {
    String url = await getAbsoluteUrl(fileName);
    AudioPlayer player = _player(mode);
    player.setReleaseMode(ReleaseMode.STOP); // this line
    await player.play(
      url,
      volume: volume,
      respectSilence: isNotification ?? respectSilence,
      stayAwake: stayAwake,
      recordingActive: recordingActive,
      duckAudio: duckAudio ?? this.duckAudio,
    );
    return player;
  }
```

This line was added because #618 . That is a good PR, but it will cause errors when not using fixedPlayer. Maybe due to large amount of `AudioPlayer` will not be released.

I think we should add a judgment like this:

```
Future<AudioPlayer> play(
    String fileName, {
    double volume = 1.0,
    bool? isNotification,
    PlayerMode mode = PlayerMode.MEDIA_PLAYER,
    bool stayAwake = false,
    bool recordingActive = false,
    bool? duckAudio,
  }) async {
    String url = await getAbsoluteUrl(fileName);
    AudioPlayer player = _player(mode);
    
    // this three lines
    if (fixedPlayer != null) {
      player.setReleaseMode(ReleaseMode.STOP);
    }
    
    await player.play(
      url,
      volume: volume,
      respectSilence: isNotification ?? respectSilence,
      stayAwake: stayAwake,
      recordingActive: recordingActive,
      duckAudio: duckAudio ?? this.duckAudio,
    );
    return player;
  }
```

If my idea meets the needs, I'm interested in working on a PR for this.

**Code to Reproduce**

```
// Create a reference
final AudioCache _audioCache = AudioCache(prefix: 'assets/audio/');

// Press several times, an error occurs
TextButton(
  onPressed: () {
    _audioCache.play('demo.mp3');
  },
  child: Text('Play'),
),
```

**Log Errors**

```
D/MediaPlayer( 5024): setSubtitleAnchor in MediaPlayer
E/MediaPlayer( 5024): error (1, -19)
E/MediaPlayer( 5024): Error (1,-19)
E/MediaPlayer( 5024): pause called in state 0, mPlayer(0x83798720)
E/MediaPlayer( 5024): error (-38, 0)
E/MediaPlayer( 5024): Attempt to perform seekTo in wrong state: mPlayer=0x83798720, mCurrentState=0
E/MediaPlayer( 5024): error (-38, 0)
E/MediaPlayer( 5024): Error (-38,0)
E/MediaPlayer( 5024): Error (-38,0)
```

**Files/URLs/Sources**

None.

**Screenshots**

None.

**Platforms**

* OS: Android
* OS version: 7.1.2
* Device: physical device. Internal model.
* flutter version: 2.0.3
* audioplayers version: 0.18.3
* release or not release: release and debug modes.
* does the error occur and does it have any peculiarities: No.

### PR DESCRIPTION
**Pull request by wenxiangjiang**
**Related Issues**

fixes #843

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]
LGTM

File: lib/audio_cache.dart [hunk 1]
```diff
@@ -158,7 +158,9 @@ class AudioCache {
   }) async {
     final uri = await load(fileName);
     final player = _player(mode);
-    player.setReleaseMode(ReleaseMode.STOP);
+    if (fixedPlayer != null) {
+      player.setReleaseMode(ReleaseMode.STOP);
```
Comment (luanpotter) [inline lib/audio_cache.dart:162]: should we set it to RELEASE?

Comment (luanpotter) [reply lib/audio_cache.dart:162]: wait RELEASE is already the default right? for new players

Comment (luanpotter) [reply lib/audio_cache.dart:162]: maybe we can just remove this entirely. but the if is definitely an improvement!

**Review by luanpotter — APPROVED** [review]
LGTM

File: lib/audio_cache.dart (hunk 1 above)
Comment (wenxiangjiang) [reply lib/audio_cache.dart:162]: @luanpotter Yes, RELEASE is the default. the if is directed at #618

**Comment by wenxiangjiang** [comment]
I think `playBytes` has the same issue.

File: lib/audio_cache.dart (hunk 1 above)
Comment (wenxiangjiang) [reply lib/audio_cache.dart:162]: Either RELEASE or STOP can solve the issue, but I think STOP may be better.

**Pull request merged**